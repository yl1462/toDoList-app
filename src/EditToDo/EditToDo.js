import React, { Component } from 'react';
import config from '../config';
import PropTypes from 'prop-types'

class EditToDo extends Component {
  state = {
    title: '',
    description: '',
    id: '',
    index: ''
  }

  componentDidMount() {
    const { title, description, id } = this.props.location.state.todo
    const { index } = this.props.location.state
    this.setState({
      title,
      description,
      id,
      index
    })
  }

  handleChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  // update the selected to do, with the given id
  handleSubmit = (e) => {
    const editedToDo = {
      title: this.state.title,
      description: this.state.description,
      id: this.state.id
  }
    e.preventDefault()
    fetch(`${config.API_ENDPOINT}/api/todo/${this.state.id}`, {
        method:'PATCH',
        headers:{
            'Content-Type':'application/json'
        },
        body:JSON.stringify(editedToDo)
    })
    .then(() => {
      console.log('edit')
      this.props.editToDo(editedToDo, this.state.index)
      this.props.history.push('/home')
    })
    .catch(err => {
        this.setState({err})
    })
}

  render() {
    const {title, description} = this.state

    return (
      <>
        {/* submit edited to do, with required title, and optional description */}
        <div className="EditToDo">
          <form onSubmit={this.handleSubmit}>
            <input 
              type='text' 
              value={title}  
              name='title'
              onChange={this.handleChange}
              className="Placeholder"
              required
              pattern="[A-Za-z0-9\~\!\@\#\$\%\^\*\(\)\_]{3}"
              title="at least 3 characters required"
              /><br />
            <textarea 
              type='text' 
              value={description}  
              name='description'
              onChange={this.handleChange}
              className="Placeholder"
              /><br />
            <button 
              type='submit' 
              >Update</button>
          </form>
        </div>
      </>
    );
  }
}

export default EditToDo;

EditToDo.propTypes = {
  history: PropTypes.object.isRequired
}