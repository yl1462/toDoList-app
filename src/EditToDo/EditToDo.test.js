import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import EditToDo from './EditToDo'

describe('renders without crashing', () => {
  //test if page loads
  it('renders without crashing', () => {
    const div = document.createElement('div')
    const props = {
      state: {
        index: 0,
        todo: {
          title: "",
          description: "",
          id: 1
        }
      }
    }

    ReactDOM.render(
      <BrowserRouter>
        <EditToDo location={props} />
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe(`EditToDo component`, () => {
  const props = {
    state: {
      index: 0,
      todo: {
        title: "",
        description: "",
        id: 1
      }
    }
  }

  //test if testing updated to do renders
  it('renders the new todo given props', () => {
    const wrapper = shallow(<EditToDo location={props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})

