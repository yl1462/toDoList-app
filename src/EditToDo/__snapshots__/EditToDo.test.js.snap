// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditToDo component renders the new todo given props 1`] = `
<Fragment>
  <div
    className="EditToDo"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="Placeholder"
        name="title"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      <br />
      <textarea
        className="Placeholder"
        name="description"
        onChange={[Function]}
        type="text"
        value=""
      />
      <br />
      <button
        type="submit"
      >
        Update
      </button>
    </form>
  </div>
</Fragment>
`;
